lib/rubocop/cop/metrics/utils/abc_size_calculator.rb

```diff
@@ -70,8 +74,28 @@ def else_branch?(node)
 
           private
 
+          def assignment?(node)
+            (node.assignment? && !assignment_doubled_in_ast?(node)) ||
+              argument?(node)
+          end
+
+          # Returns true for nodes which otherwise would be counted
+          # as one too many assignment
+          def assignment_doubled_in_ast?(node)
+            node.masgn_type? || node.or_asgn_type? || node.and_asgn_type?
+          end
+
           def branch?(node)
-            BRANCH_NODES.include?(node.type)
+            BRANCH_NODES.include?(node.type) || block_with_body?(node)
```

Comment:
I assume this is what is causing the large number of new failures?

Reply:
That count's +1 for (non-empty) blocks. I've yet to check for cyclomatic complexity but a block is a disconnected component in the graph so +1. That adds a bunch for sure. Also block arguments, so `[1, 2, 3, 4, 5].tap {|a, b, c, d, e| #... }` goes from `<0, 1, 0>+...` to `<5, 3, 0>+...`.
There are also a lot of `option[:foo] = :bar` that were not counted as assignments before...

Reply:
I've amended my PR to only count block arguments.